File: src/ee/ads/NullRewardedAd.cpp
//
//  NullRewardedVideo.cpp
//  ee_x
//
//  Created by Zinge on 10/27/17.
//
//

#include "ee/ads/NullRewardedAd.hpp"

#include <ee/coroutine/Task.hpp>

namespace ee {
namespace ads {
using Self = NullRewardedAd;

void Self::destroy() {}

bool Self::isLoaded() const {
    return false;
}

Task<bool> Self::load() {
    co_return false;
}

Task<IRewardedAdResult> Self::show() {
    co_return IRewardedAdResult::Failed;
}
} // namespace ads
} // namespace ee

File: unity/Runtime/Ads/NullRewardedAd.cs
using System;
using System.Threading.Tasks;

namespace EE {
    public class NullRewardedAd : ObserverManager<IRewardedAdObserver>, IRewardedAd {
        public void Destroy() {
            throw new NotImplementedException();
        }

        public bool IsLoaded { get; } = false;

        public Task<bool> Load() {
            return Task.FromResult(false);
        }

        public Task<IRewardedAdResult> Show() {
            return Task.FromResult(IRewardedAdResult.Failed);
        }
    }
}